src/routes/+layout.svelte

```diff
@@ -13,19 +15,28 @@
 		['/settings', 'Settings']
 	];
 
-	let theme = 'light';
+	$: theme = $settingsStore?.userTheme;
+
+	onMount(() => {
+		if (browser && !theme) {
+			const systemTheme = window.matchMedia('(prefers-color-scheme: dark)').matches
+				? 'dark'
+				: 'light';
+			settingsStore.update((store) => ({
+				...store!,
+				userTheme: systemTheme
+			}));
+		}
+	});
 
 	function toggleTheme() {
 		theme = theme === 'light' ? 'dark' : 'light';
 		document.documentElement.setAttribute('data-color-theme', theme);
+		settingsStore.update((store) => ({
+			...store!,
+			userTheme: theme
+		}));
```

Comment:
This is not necessarily wrong, but one thing to note is that inside a Svelte component you don't need to call `settingsStore.update()`, you can simply do:

```ts
$settingsStore!.userTheme = systemTheme;
```

> Assignments to $-prefixed variables require that the variable be a writable store, and will result in a call to the store's .set method. — https://svelte.dev/docs/svelte-components#script-4-prefix-stores-with-$-to-access-their-values

Also, I'm not sure what the best way to handle this is but if `$settingsStore == null` (which I don't know if it's even possible in this case), calling `$settingsStore!` would throw an error to the console, right? Which I guess it's useful because we are always expecting the store to be present not matter what.

[In other places](https://github.com/fmaclen/hollama/blob/main/src/routes/sessions/%5Bid%5D/%2Bpage.svelte#L59) I added a conditional check, but maybe we actually need to intentionally throw an error if we really are expecting `settingsStore` to be set at that point.

I personally almost never use the syntax `variable!`, feels a bit like a code smell. My gut tells me that in both `onMount()` and `toggleTheme()` we'd be better off adding a guard clause, specially `onMount()` since running the rest of the code in that section doesn't make sense if the store isn't present for whatever reason.

What do you think?

Reply:
Maybe something like this can make the code a bit more readable?

```ts
onMount(() => {
	if (!$settingsStore || !browser || theme) return;
	$settingsStore!.userTheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
});
```

```ts
function toggleTheme() {
	theme = theme === 'light' ? 'dark' : 'light';
	document.documentElement.setAttribute('data-color-theme', theme);
	if ($settingsStore) $settingsStore.userTheme = theme;
}
```

Reply:
Nice! I agree. I don't like using `variable!` either, but I also thought about the need of `settingsStore` to always be present.

I've just commited this, so I am merging now.